File: PHP/876. Middle of the Linked List.php
<?php

/**
 *  @noinspection PhpIllegalPsrClassPathInspection, PhpMultipleClassDeclarationsInspection, PhpMultipleClassesDeclarationsInOneFile
 */
class ListNode
{
    public mixed $val = 0;

    public mixed $next = null;

    public function __construct($val = 0, $next = null)
    {
        $this->val = $val;
        $this->next = $next;
    }
}

class Solution
{
    /**
     * @noinspection PhpUnused
     */
    public function middleNode(?ListNode $head): ?ListNode
    {
        $slow = $fast = $head;
        while ($fast && $fast->next) {
            [$slow, $fast] = [$slow->next, $fast->next->next];
        }

        return $slow;
    }
}

File: Python/876. Middle of the Linked List.py
from typing import Optional


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


# noinspection PyMethodMayBeStatic
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        slow = fast = head
        while fast and fast.next:
            slow, fast = slow.next, fast.next.next
        return slow
